lib/cretonne/meta/isa/intel/recipes.py

```diff
@@ -160,6 +160,11 @@ def __call__(self, *ops, **kwargs):
                 instp=self.instp,
                 isap=self.isap,
                 emit=replace_put_op(self.emit, name))
+
+            recipe.ins = tuple(GPR8 if (reg is GPR) else reg for reg in tuple(recipe.ins))
+            recipe.outs = tuple(GPR8 if (reg is GPR) else reg for reg in tuple(recipe.outs))
```

Comment:
Yes, exactly, but you should also map FPR -> FPR8. And perhaps move the list comprehension into its own function?